pkgs/applications/virtualization/qemu/default.nix

```
@@ -170,7 +170,9 @@ stdenv.mkDerivation (finalAttrs: {
     perl
 
     # Don't change this to python3 and python3.pkgs.*, breaks cross-compilation
-    (python3Packages.python.withPackages (ps: with ps; [ distlib ]))
+    # And don't use `python.withPackages` directly, use `python.pythonOnBuildForHost.withPackages` instead
+    # because the result of `withPackages` doesn't get respliced
+    (python3Packages.python.pythonOnBuildForHost.withPackages (ps: with ps; [ distlib ]))
```

```suggestion
    python3Packages.python
    python3Packages.distlib
```

Should fix cross *and* qemu-user

---

This is effectively a revert of https://github.com/NixOS/nixpkgs/commit/e651d115d5f2a6d37661578370993d345d97c71c and then addition of `python3Packages.distlib`. Putting it here for posterity.